wfe2/wfe_test.go

```diff
@@ -4138,3 +4154,81 @@ func TestNewOrderWithProfile(t *testing.T) {
 	test.AssertEquals(t, errorResp2["type"], "urn:ietf:params:acme:error:malformed")
 	test.AssertEquals(t, errorResp2["detail"], "Invalid certificate profile, \"test-profile\": not a recognized profile name")
 }
+
+func makeARICertID(leaf *x509.Certificate) (string, error) {
+	if leaf == nil {
+		return "", errors.New("leaf certificate is nil")
+	}
+
+	// Marshal the Serial Number into DER.
+	der, err := asn1.Marshal(leaf.SerialNumber)
+	if err != nil {
+		return "", err
+	}
+
+	// Check if the DER encoded bytes are sufficient (at least 3 bytes: tag,
+	// length, and value).
+	if len(der) < 3 {
```

Comment:
Would it be useful to roundtrip the DER bytes with `x509.ParseCertificate(der)` and compare the result to `leaf`?

Reply:
The der here is not a whole certificate, it's just a serial number, so x509.ParseCertificate would barf on it. For the purposes of this test, this is the right check -- we just need to make sure it's at least length 3 so that the `[2:]` in the next stanza doesn't try to index out-of-bounds and panic.

Reply:
Gotcha, thank you.